### PR TITLE
Fix for `src-files` not being used when specified in a config file

### PR DESCRIPTION
**Pull request by csalerno-asml**
This PR fixes https://github.com/jazzband/pip-tools/issues/2006

`src-files` is found in the click context, `ctx`, but however is not populating the variable `src_files`, so that it is an empty tuple.
For this reason, the tool will then start to select a requirement file, as shown in the linked issue.

##### Contributor checklist

- [x] Included tests for the changes.
- [x] PR title is short, clear, and ready to be included in the user-facing changelog.

##### Maintainer checklist

- [x] Verified one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

**Comment by webknjaz** [comment]
@csalerno-asml thanks for starting this! This PR would need a regression test to be mergeable. Could you add one?

**Comment by csalerno-asml** [comment]
Done @webknjaz :)

File: src/pip-tools
Comment (chrysle) [inline src/pip-tools:1]: Why did you add this?

File: piptools/scripts/compile.py
Comment (chrysle) [inline piptools/scripts/compile.py:169]: We generally use RST syntax, I think consistency with the comments would be good.

```suggestion
    # If ``src-files` was not provided as an input, but rather as config,
    # it will be part of the click context ``ctx``.
    # However, is ``src_files`` is specified, then we want to use that.
```

File: piptools/scripts/compile.py
Comment (chrysle) [inline piptools/scripts/compile.py:170]:
```suggestion
    if not src_files and "src_files" in ctx.default_map:
```

I think that is sure to exist, have you tested?

File: tests/test_cli_compile.py
Comment (chrysle) [inline tests/test_cli_compile.py:3462]:
```suggestion
    req_in.write_text("small-fake-a==0.1", encoding="utf-8")
    foo_in.write_text("small-fake-b==0.1", encoding="utf-8")
```

File: tests/test_cli_compile.py
Comment (chrysle) [inline tests/test_cli_compile.py:3483]:
```suggestion
    req_in.write_text("small-fake-a==0.1", encoding="utf-8")
    foo_in.write_text("small-fake-b==0.1", encoding="utf-8")
```

File: tests/test_cli_compile.py
Comment (chrysle) [inline tests/test_cli_compile.py:3487]:
```suggestion
    assert out.exit_code == 0, out
```

File: tests/test_cli_compile.py
Comment (chrysle) [inline tests/test_cli_compile.py:3466]:
```suggestion
    assert out.exit_code == 0, out
```

File: src/pip-tools
Comment (csalerno-asml) [reply src/pip-tools:1]: addressed, was added by mistake

**Review by chrysle — APPROVED** [review]
LGTM, but I'd like to hold this back for now per https://github.com/jazzband/pip-tools/issues/2006#issuecomment-1793446717.

File: tests/test_cli_compile.py
Comment (atugushev) [inline tests/test_cli_compile.py:3465]: Could you please check the whole output in `out.stderr` (here and in the other test)? See `test_not_specified_input_file` as a reference.

Comment (csalerno-asml) [reply tests/test_cli_compile.py:3465]: I tried to stick to what other tests did, and also this is enough, since I am showing that one of the two requirements file is ignored

**Comment by csalerno-asml** [comment]
@chrysle fine but this PR is open since August https://github.com/jazzband/pip-tools/pull/1966 
This one can be merged as-is and then reworked if necessary - or the other one gets a bump

**Comment by csalerno-asml** [comment]
Any update @chrysle @atugushev @webknjaz ?

**Comment by webknjaz** [comment]
@csalerno-asml #1966 is approved and will be merged once the CI gets unblocked (it's a repo-global blocker).

**Comment by chrysle** [comment]
Since that's now merged, could you rebase?

**Comment by webknjaz** [comment]
That's not a rebase, though.

**Comment by csalerno-asml** [comment]
@webknjaz I rebased, but then there was a merge conflict (don't know why), so I just merged `main` inside this branch. There are no conflicts however, so I don't see it as an issue

**Comment by webknjaz** [comment]
Well, this will cause a "foxtrot merge".

**Comment by csalerno-asml** [comment]
@webknjaz re-rebased

**Comment by chrysle** [comment]
Thanks!

**Comment by csalerno-asml** [comment]
hey @chrysle , do you know when this fix will be released? do you guys have a schedule, or a release strategy?

**Comment by webknjaz** [comment]
Not really. There was some shared repo where releases could be requested, I think.

**Comment by csalerno-asml** [comment]
From what I see, this is done manually? https://github.com/jazzband/pip-tools/pull/1958

**Comment by atugushev** [comment]
Let's make a release. I'll do it today.

**Comment by atugushev** [comment]
Released as part of pip-tools `v7.4.0`.

**Pull request merged**
